<launch>
	<remap from="/front/pose/3d" to="/pose/3d" />
	<remap from="/back/pose/3d" to="/pose/3d" />

	<!-- Left image detector node -->
	<node pkg="rovi2_development" type="image_detector" name="image_detector_left_back" output="screen">
		<param name="image_sub" value="/back/camera/left/image_raw" />
		<param name="point_pub" value="/back/pose/2d_left" />
		<param name="calibration_yaml_path" value="back/left.yaml" />
	</node>

	<!-- Right image detector node -->
	<node pkg="rovi2_development" type="image_detector" name="image_detector_right_back" output="screen" >
		<param name="image_sub" value="/back/camera/right/image_raw" />
		<param name="point_pub" value="/back/pose/2d_right" />
		<param name="calibration_yaml_path" value="back/right.yaml" />
	</node>

	<!-- Stereo node -->
	<node pkg="rovi2_development" type="stereo" name="stereo_node_back" output="screen">
		<param name="point_sub_left" value="/back/pose/2d_left" />
		<param name="point_sub_right" value="/back/pose/2d_right" />
		<param name="point_pub" value="/back/pose/3d" />
                <param name="calibration_yaml_path_left" value="back/left.yaml" />
                <param name="calibration_yaml_path_right" value="back/right.yaml" />
		<param name="param_hte_path" value="/back/hte_back.yaml" />
	</node>

	<node pkg="rovi2_development" type="image_detector" name="image_detector_left_front" output="screen">
		<param name="image_sub" value="/front/camera/left/image_raw" />
		<param name="point_pub" value="/front/pose/2d_left" />
		<param name="calibration_yaml_path" value="front/left.yaml" />
	</node>

	<!-- Right image detector node -->
	<node pkg="rovi2_development" type="image_detector" name="image_detector_right_front" output="screen" >
		<param name="image_sub" value="/front/camera/right/image_raw" />
		<param name="point_pub" value="/front/pose/2d_right" />
		<param name="calibration_yaml_path" value="front/right.yaml" />
	</node>

	<!-- Stereo node -->
	<node pkg="rovi2_development" type="stereo" name="stereo_node_front" output="screen">
		<param name="point_sub_left" value="/front/pose/2d_left" />
		<param name="point_sub_right" value="/front/pose/2d_right" />
		<param name="point_pub" value="/front/pose/3d" />
                <param name="calibration_yaml_path_left" value="front/left.yaml" />
                <param name="calibration_yaml_path_right" value="front/right.yaml" />
		<param name="param_hte_path" value="/front/hte_front.yaml" />
                <param name="flip_z" value="true" />
	</node>
	<node pkg="rovi2_development" type="3DPoseLogger.py" name="PoseLoggerBack">
		<param name="sub_3d_point" value="/back/pose/3d" />
		<param name="filename" value="/tmp/3dpose_back.log" />
	</node>

	<node pkg="rovi2_development" type="3DPoseLogger.py" name="PoseLoggerFront">
		<param name="sub_3d_point" value="/front/pose/3d" />
		<param name="filename" value="/tmp/3dpose_front.log" />
	</node>

</launch>
